### PR TITLE
[docs] Replace Truffle/web3 mentions and sample-project with Waffle/ehters

### PR DESCRIPTION
**Pull request by viarnes**
- [x] Mentions and links to Waffle/web3 replaced with Truffle/ethers
- [x] Code samples changed (config, task, test and script
- [x] Change `npx buidler` sample project files and code generation
  - [x]  Replace msg: Do you want to install the sample project's dependencies with npm (@nomiclabs/buidler-truffle5 @nomiclabs/buidler-web3 web3@^1.2.0)? with Waffle + Ethers + Chai
  - [x]  buidler.config.js
  - [x]  sample-test.js
  - [x]  sample-script.js

### CODE REVIEW

**Comment by tmilar** [comment]
We should also update [Typescript Support](https://buidler.dev/guides/typescript.html) page to match the getting started config for ethers/waffle as well, right? Currently, it's written for truffle/web3

File: packages/buidler-core/sample-project/scripts/sample-script.js [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-  const Greeter = env.artifacts.require("Greeter");
-  const greeter = await Greeter.new("Hello, world!");
+  await greeter.deployed();
 
-  console.log("Greeter address:", greeter.address);
+  console.log("Greeter deployed to:", greeter.address);
 }
 
-// We recommend this pattern to be able to use async/await everywhere
-// and properly handle errors.
```
Comment (fzeoli) [inline packages/buidler-core/sample-project/scripts/sample-script.js:19]: why remove this?

File: packages/buidler-core/sample-project/test/sample-test.js [hunk 2]
```diff
@@ -1,34 +1,14 @@
...
-
-// Vanilla Mocha test. Increased compatibility with tools that integrate Mocha.
-describe("Greeter contract", function() {
-  let accounts;
-
-  before(async function() {
-    accounts = await web3.eth.getAccounts();
-  });
-
-  describe("Deployment", function() {
```
Comment (fzeoli) [inline packages/buidler-core/sample-project/test/sample-test.js:23]: why did you remove this test rather than rewrite with ethers?

**Comment by viarnes** [comment]
I worked on two different branches (`replace-truffle-web3-with-waffle-ethers` and `replace-truffle-web3-with-waffle-ethers-full-doc`) due to a misunderstood but ended up merging them into the first one.

File: packages/buidler-core/sample-project/scripts/sample-script.js (hunk 1 above)
Comment (viarnes) [reply packages/buidler-core/sample-project/scripts/sample-script.js:19]: I asked @alcuadrado to remove this in order to simplify the script. There are two main reasons:

1. Extending the BRE is an advanced use and is explained in `/advanced/buidler-runtime-environment.html`
2. When running `npx buidler run scripts/sample-script.js` the contracts were compiling twice.

File: packages/buidler-core/sample-project/test/sample-test.js (hunk 2 above)
Comment (viarnes) [reply packages/buidler-core/sample-project/test/sample-test.js:23]: I rewrited the test using ethers but I removed one of them because it was using TruffleContract (`contract` instead of Mocha's `describe`). Also, both tests were doing the same and `accounts` wasn't being used.

File: docs/guides/typescript.md [hunk 3]
```diff
@@ -44,16 +44,16 @@ mv buidler.config.js buidler.config.ts
 We also need to adapt it to explicitly import the Buidler config DSL, and use the [Buidler Runtime Environment] explicitly.
 
 For example, the sample project's config turns from this
-```js{1,6,13}
-usePlugin("@nomiclabs/buidler-truffle5");
+```js{1,5,13}
```
Comment (fzeoli) [inline docs/guides/typescript.md:47]: first line here is highlighted in there's no changes to it from one example to the next

File: docs/guides/typescript.md [hunk 4]
```diff
@@ -119,36 +119,36 @@ export default config;
...
 
 ## Writing tests and scripts
 
-To write your smart contract tests and scripts you'll most likely need access to an Ethereum library to interact with your smart contracts. This will probably be one of [buidler-truffle5](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-truffle5), [buidler-web3](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-web3) or [buidler-ethers](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-web3), all of which inject instances into the [Buidler Runtime Environment].
+To write your smart contract tests and scripts you'll most likely need access to an Ethereum library to interact with your smart contracts. This will probably be one of [buidler-waffle](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-waffle) & [buidler-ethers](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-ethers) or [buidler-truffle](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-truffle) & [buidler-web3](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-web3), all of which inject instances into the [Buidler Runtime Environment].
```
Comment (fzeoli) [inline docs/guides/typescript.md:138]: waffle isn't a library to interact with smart contracts

File: packages/buidler-core/sample-project/scripts/sample-script.js [hunk 5]
```diff
@@ -1,22 +1,13 @@
-// We require the Buidler Runtime Environment explicitly here. This is optional
```
Comment (fzeoli) [inline packages/buidler-core/sample-project/scripts/sample-script.js:1]: I guess this is the bit you were referring to with your comment about extending the BRE, but I don't get it. This isn't an extension, it's just plain usage and this worked as an example of how to use it directly, which is basic Buidler usage. Why remove it?

File: packages/buidler-core/sample-project/scripts/sample-script.js (hunk 1 above)
Comment (viarnes) [reply packages/buidler-core/sample-project/scripts/sample-script.js:19]: My bad. I copied the script from the tutorial, should have checked the comments before commiting.

Re-adding those lines.

File: docs/getting-started/README.md [hunk 6]
```diff
@@ -51,15 +51,13 @@ $ npx buidler
...
-tests built for Truffle. You can learn more about it [in this guide](../guides/truffle-testing.md). 
-For now, all you need to know is that you may need to install some dependencies with
- 
-```bash
-npm install --save-dev @nomiclabs/buidler-truffle5 @nomiclabs/buidler-web3 web3
-```
+The sample project will ask you to install `buidler-waffle` and `buidler-ethers`, which makes Buidler compatible with tests built for Waffle. You can learn more about it [in this guide](../guides/waffle-testing.md). 
+
+::: tip
+You can install those dependencies manually with `npm install --save-dev @nomiclabs/buidler-waffle ethereum-waffle chai @nomiclabs/buidler-ethers ethers`
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/getting-started/README.md:59]: What I was trying to saying was:

In case you missed the installation of those depencies you can install them manually with...

File: docs/guides/scripts.md [hunk 7]
```diff
@@ -13,8 +13,8 @@ as global variables.
 
 These scripts must be run through Buidler: `npx buidler run script.js`. 
 
-This makes it easy to port scripts that were developed for Truffle, which follows this approach,
-by using the [buidler-truffle5](https://github.com/nomiclabs/buidler/tree/master/packages/buidler-truffle5). 
+This makes it easy to port scripts that were developed for other tools such as Waffle, which follows this approach,
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:16]: Should I mention Truffle here? Also we could end the sentence on "tools": _This makes it easy to port scripts that were developed for other tools_.

File: docs/guides/scripts.md [hunk 8]
```diff
@@ -40,20 +40,22 @@ drwxr-xr-x    3 fzeoli  staff      96 Jul 30 15:27 scripts
...
+Inside `scripts/` you will find `sample-script.js`. Add the highlighted lines to it:
+
+```js{1-2,5}
 const bre = require("@nomiclabs/buidler");
+const ethers = bre.ethers;
 
 async function main() {
-  // You can run Buidler tasks from a script.
-  // For example, we make sure everything is compiled by running "compile"
   await bre.run("compile");
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:50]: What makes the script standalone? There are 3 lines being highlighted, am I missing something?

File: docs/guides/scripts.md [hunk 9]
```diff
@@ -40,20 +40,22 @@ drwxr-xr-x    3 fzeoli  staff      96 Jul 30 15:27 scripts
...
-Inside `scripts/` you will find `sample-script.js`:
-```js
+Inside `scripts/` you will find `sample-script.js`. Add the highlighted lines to it:
+
+```js{1-2,5}
 const bre = require("@nomiclabs/buidler");
+const ethers = bre.ethers;
 
 async function main() {
-  // You can run Buidler tasks from a script.
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:48]: Re-adding the comments

File: docs/guides/scripts.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:50]: Oh, I think I see your point here. It reads as those lines are responsible for making the script standalone, not in fact that the BRE is being extended. I'll fix it by adding some extra information/comments.

File: docs/guides/scripts.md [hunk 10]
```diff
@@ -62,24 +64,30 @@ main()
...
-$ npx buidler run scripts/sample-script.js
+$ npx buidler run scripts/sample-script.js 
+All contracts have already been compiled, skipping compilation.
 All contracts have already been compiled, skipping compilation.
-Greeter address: 0x494d39079b81c620c0ebea503b9295331bfc34c2
+Greeter address: 0x7c2C195CD6D34B8F845992d380aADB2730bB9C6F
 ```
 
+::: tip
+Did you notice the double compile message? When running a script through `npx buidler run`, the `compile` task will be called before running the script, but you can skip this with the `--no-compile` parameter.
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:88]: I wouldn't remove `bre.run('compile')` because it's used as example on how to use Buidler as a library making the script standalone.  Maybe we could pick another `task` instead of `compile` to use in the sample script. 

I honestly prefer `compile` + the `::: tip :::`.

File: docs/guides/truffle-testing.md [hunk 11]
```diff
@@ -13,28 +14,92 @@ npm init --yes
 npm install --save-dev @nomiclabs/buidler
 ```
 
-Now run `npx buidler` inside your project folder and create a sample project. This is what the file structure should look like once you're done:
+Now run npx buidler inside your project folder and select `Create an empty buidler.config.js`.
+
+Let's now install `web3.js`, `Truffle` and their Buidler plugins, which will allow Truffle to use Buidler EVM and get stack traces and `console.log` functionality.
```
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/truffle-testing.md:19]: I copy that phrase from [here](https://buidler.dev/guides/waffle-testing.html):

"Done? Great. Let's now install ethers.js, Waffle and their Buidler plugins, which will allow Waffle to use Buidler EVM and get stack traces and console.log functionality."

How should I change it?

File: packages/buidler-core/sample-project/scripts/sample-script.js (hunk 5 above)
Comment (viarnes) [reply packages/buidler-core/sample-project/scripts/sample-script.js:1]: As I mentioned before, I asked @alcuadrado if I could remove this in order to simplify the script. There are two main reasons:

- Extending the BRE is an advanced use and is explained in /advanced/buidler-runtime-environment.html
- When running `npx buidler run scripts/sample-script.js` the contracts were compiling twice.

File: docs/getting-started/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/getting-started/README.md:59]: let's change it to

>Buidler will let you know how, but in case you missed it you can install them with `npm install --save-dev @nomiclabs/buidler-waffle ethereum-waffle chai @nomiclabs/buidler-ethers ethers`

File: docs/guides/scripts.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/scripts.md]: I like that, yeah. We can even add a bit more information:
"This makes it easy to port scripts that were developed for other tools that inject variables into the global state."

File: docs/guides/scripts.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/scripts.md:52]: yeah, but compile runs anyway, so that line doesn't make it standalone

File: docs/guides/scripts.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/scripts.md:90]: I see that, but invoking the `compile` task to yield two compilations and then tell the user to use the `--no-compile` arg to undo our `compile` call by disabling the default behavior of compiling is just really convoluted.

@alcuadrado any thoughts?

File: docs/guides/truffle-testing.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/truffle-testing.md:19]: `Let's now install the Truffle and Web3.js plugins, as well as web3.js itself`

File: docs/guides/scripts.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md:52]: Below the script:

> By accessing the [Buidler Runtime Environment](#) at the top, you are now allowed to run the script in a standalone fashion. Lines 2 and 7 are just examples on how to use it. Let's re-run the script with `node`:

[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/scripts.md:52]: Okay, I was misunderstanding everything. Forget my initial comment here and let's just explain how these highlighted lines make it standalone after the code snipper. Basically explain that:

- the BRE components are accessed explicitly through the BRE rather than using from a global scope injection that's only available when running through Buidler
- the compile task is invoked manually. When Buidler runs a script through `buidler run` compile is run before, so to make sure everything is compiled when running directly through node.js we call `compile`

File: docs/guides/scripts.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (alcuadrado) [reply docs/guides/scripts.md:91]: Why not just leave the `bre.run` as a comment?

[the remark this answers is not in the document]
Comment (alcuadrado) [reply docs/guides/scripts.md:91]: It works as an explanation and doesn't do the double compilation thing.

[the remark this answers is not in the document]
Comment (fzeoli) [reply docs/guides/scripts.md:91]: because the line is needed for the standalone script. 
- leave the call to `bre.run` in and simply add a step to comment it out before running the script through Buidler, explaining that that's not needed when running scripts through buidler
- remove the tip block

[the remark this answers is not in the document]
Comment (viarnes) [reply docs/guides/scripts.md]: I removed the tip block but left the call to `bre.run` commented because the script is also mentioned on the Getting Started section. Also added a step to uncomment this line out to explain how to manually compile.
